[bug] CMakeConfigDeps: package_framework does not have IMPORTED_CONFIGURATIONS list specified

### Describe the bug

Here is a part of recipe's conanfile.py that I use
```
def package_info(self):
        self.cpp_info.set_property("cmake_target_name", "amnezia::openvpnadapter")
        self.cpp_info.type = PackageType.STATIC
        self.cpp_info.package_framework = True
        self.cpp_info.location = os.path.join(self.package_folder, "OpenVPNAdapter.framework")
```

And it generates 4 files called openvpnadapter-Targets-{config}.cmake with output like this (e.g. DEBUG):
```
set_target_properties(amnezia::openvpnadapter PROPERTIES
    IMPORTED_LOCATION_DEBUG "${openvpnadapter_PACKAGE_FOLDER_DEBUG}/OpenVPNAdapter.framework"
    FRAMEWORK TRUE)
if(CMAKE_VERSION VERSION_LESS "3.24")
    set_property(TARGET amnezia::openvpnadapter APPEND PROPERTY INTERFACE_COMPILE_OPTIONS
                 $<$<COMPILE_LANGUAGE:CXX>:-F${openvpnadapter_PACKAGE_FOLDER_DEBUG}/>)
    set_property(TARGET amnezia::openvpnadapter APPEND PROPERTY INTERFACE_COMPILE_OPTIONS
                 $<$<COMPILE_LANGUAGE:C>:-F${openvpnadapter_PACKAGE_FOLDER_DEBUG}/>)
endif()
```

The problem is that this configuration does not have any `IMPORTED_CONFIGURATIONS` set. So when I try to use `XCODE_LINK_BUILD_PHASE_MODE KNOWN_LOCATION` option on my original target with this library linked:
```
set_target_properties(networkextension PROPERTIES XCODE_LINK_BUILD_PHASE_MODE KNOWN_LOCATION)
find_package(openvpnadapter REQUIRED)
target_link_libraries(networkextension PRIVATE amnezia::openvpnadapter)
```
The issue appears:
```
[cmake] CMake Error in client/ios/networkextension/CMakeLists.txt:
[cmake]   IMPORTED_LOCATION not set for imported target "amnezia::openvpnadapter".
```

I did not find any way to workaround that behavior in recipe itself (or maybe I do actually use those functionality inproperly?)
But setting this property manually resolves the issue
```
set_property(TARGET amnezia::openvpnadapter APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)
```

I would suggest putting IMPORTED_CONFIGURATIONS thingy in the template for `package_framework`, in the way it happens with libraries. Should not break anything IMO
Please feel free to ask for a full files/configurations - I will be happy to share any details

### How to reproduce it

_No response_

--- a/conan/tools/cmake/cmakeconfigdeps/target_configuration.py
+++ b/conan/tools/cmake/cmakeconfigdeps/target_configuration.py
@@ -469,6 +469,7 @@ class TargetConfigurationTemplate2:
                      "{{config_wrapper(config, lib_info["frameworks"])}}")
         {% endif %}
         {% if lib_info.get("package_framework") %}
+        set_property(TARGET {{lib}} APPEND PROPERTY IMPORTED_CONFIGURATIONS {{config}})
         set_target_properties({{lib}} PROPERTIES
             IMPORTED_LOCATION_{{config}} "{{lib_info["package_framework"]["location"]}}"
             FRAMEWORK TRUE)

--- a/test/integration/toolchains/cmake/cmakeconfigdeps/test_cmakeconfigdeps_frameworks.py
+++ b/test/integration/toolchains/cmake/cmakeconfigdeps/test_cmakeconfigdeps_frameworks.py
@@ -83,7 +83,7 @@ def test_framework_add_imported_configurations():
         def package_info(self):
             self.cpp_info.set_property("cmake_target_name", "frame::frame")
             self.cpp_info.type = "static-library"
-            self.cpp_info.package_framework = True
+            self.cpp_info.package_framework = "Frame"
             self.cpp_info.location = os.path.join(self.package_folder, "Frame.framework")
     """)
     tc = TestClient()

--- a/test/integration/toolchains/cmake/cmakeconfigdeps/test_cmakeconfigdeps_frameworks.py
+++ b/test/integration/toolchains/cmake/cmakeconfigdeps/test_cmakeconfigdeps_frameworks.py
@@ -65,3 +65,30 @@ def test_framework_only_component_generates_target():
     tc.run(f"install --requires=frame/1.0 -g CMakeConfigDeps")
     targets = tc.load("frame-Targets-release.cmake")
     assert "add_library(frame::frame INTERFACE IMPORTED)" in targets
+
+
+def test_framework_add_imported_configurations():
+    """
+    Issue: https://github.com/conan-io/conan/issues/20034
+    """
+    conanfile = textwrap.dedent(f"""
+    import os
+    from conan import ConanFile
+
+    class MyFramework(ConanFile):
+        name = "frame"
+        version = "1.0"
+        settings = "os", "arch", "compiler", "build_type"
+
+        def package_info(self):
+            self.cpp_info.set_property("cmake_target_name", "frame::frame")
+            self.cpp_info.type = "static-library"
+            self.cpp_info.package_framework = True
+            self.cpp_info.location = os.path.join(self.package_folder, "Frame.framework")
+    """)
+    tc = TestClient()
+    tc.save({"conanfile.py": conanfile})
+    tc.run("create -s build_type=Debug")
+    tc.run(f"install --requires=frame/1.0 -g CMakeConfigDeps -s build_type=Debug")
+    targets = tc.load("frame-Targets-debug.cmake")
+    assert "set_property(TARGET frame::frame APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)" in targets

